src/data/polynomial/algebra_map.lean

```diff
@@ -217,16 +217,23 @@ theorem eval_unique (φ : R[X] →ₐ[R] A) (p) :
   φ p = eval₂ (algebra_map R A) (φ X) p :=
 by rw [← aeval_def, aeval_alg_hom, aeval_X_left, alg_hom.comp_id]
 
+theorem aeval_alg_hom_class_apply {F : Type*} [alg_hom_class F R A B] (f : F) (x : A) (p : R[X]) :
```

Comment:
I'm not sure what the preferred style is. But another option would be to simply generalize the existing `aeval_alg_hom_apply` without creating a separate lemma.
```suggestion
theorem aeval_alg_hom_apply {F : Type*} [alg_hom_class F R A B] (f : F) (x : A) (p : R[X]) :
```

Reply:
@Vierkantor What is the preferred style for such a generalization?

Reply:
I'd go for "simply generalize the existing `aeval_alg_hom_apply` without creating a separate lemma". As for the name, usually I don't add the `_class` even if it's generic like here. (The reasoning is that all elements of an instance of `alg_hom_class` *are* algebra homomorphisms, plus optional extra hypotheses.)